lib/game_config.rb

```diff
@@ -7,48 +7,53 @@ def initialize (questions, parsed_arguments = nil)
     @cpu_difficulty_player_one = :not_set
     @cpu_difficulty_player_two = :not_set
     @questions = questions
+    @parsed_arguments = parsed_arguments
 
-    if parsed_arguments.color == nil
-      configure_color_board()
+    configure_color_board()
+    configure_all_players()
+  end
+
+
+  def configure_color_board
+    if @parsed_arguments.color == nil
+      @color = @questions.yes_or_no?("COLOR?:")
     else
-      @color = parsed_arguments.color 
+      @color = @parsed_arguments.color 
     end
+  end
 
-    if parsed_arguments.game_type == nil
-      configure_game_type()
+  def configure_all_players()
+    #player 1
+    if @parsed_arguments.player_one_type == nil
+      configure_player_type(:player_one)
+    else
+      @player_one_type = @parsed_arguments.player_one_type
+      @cpu_difficulty_player_one = @parsed_arguments.cpu_difficulty_player_one
+    end
+    #player 2
+    if @parsed_arguments.player_two_type == nil
+      configure_player_type(:player_two)
     else
-      @player_one_type = set_player_type(parsed_arguments, :player_one)
-      @player_two_type = set_player_type(parsed_arguments, :player_two)
+      @player_two_type = @parsed_arguments.player_two_type
+      @cpu_difficulty_player_two = @parsed_arguments.cpu_difficulty_player_two
     end
   end
 
-  def set_player_type(parsed_arguments, player)
+
+  def configure_player_type(player)
```

Comment:
This is a good place to start thinking about refactoring. For example, the  computer if statement could probably be moved out and made into a method. But it isn't a blocker for moving forward.

Reply:
Understood, thank you